src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -1030,6 +1035,52 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
             'value': frontend
         })
 
+        mgr_map = self.mgr.get('mgr_map')
+        if 'prometheus' in mgr_map.get('services', {}):
```

Comment:
I discussed this PR with @cbodley. Could we remove this assumption that the caches are always configured if the `prometheus` service is in up? We don't want the caches ever being enabled without users explicitly configuring them.

Reply:
Ok cool! I made it general for any monitoring user but yes this also makes sense. We should somehow advice user to create rgw service creation with or w/o user/bucket cache as per their use case, probably somewhere in doc cephadm docs? wdyt @alimaredia @adk3798